src/rrd_cgi.c

```diff
@@ -560,13 +559,13 @@ static char *rrdsetenv(
     const char **args)
 {
     if (argc >= 2) {
-        const size_t len = strlen(args[0]) + strlen(args[1]) + 2;
-        char *xyz = (char *) malloc(len);
+        char *xyz = NULL;
 
+        asprintf(&xyz, "%s=%s", args[0], args[1]);
```

Comment:
But we _never_ literally call `rrd_asprintf`, only `asprintf`.  The shim includes some macro magic.  I didn't verify what the magic was doinng, I assumed it did the only sensible thing.